client/pages/saved-item-details.jsx

```diff
@@ -0,0 +1,195 @@
+import NotFound from './not-found';
+
+export default class SavedItemDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      listing: null,
+      sellerEmail: null,
+      formActive: false,
+      saved: false,
+      savedPrompt: 'saved-prompt saved-hidden',
+      savedPromptText: 'Saved'
+    };
+    this.handleContactButton = this.handleContactButton.bind(this);
+    this.handleCancelButton = this.handleCancelButton.bind(this);
+    this.handleSaveButton = this.handleSaveButton.bind(this);
+    this.prompt = this.prompt.bind(this);
+  }
+
+  componentDidMount() {
+    fetch(`/api/user/saved/listing/${this.props.listingId}`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Access-Token': this.props.token
+      }
+    })
+      .then(res => res.json())
+      .then(listing => this.setState({ listing }));
+
+    fetch(`/api/user/saved/${this.props.listingId}`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-Access-Token': this.props.token
+      }
+    })
+      .then(res => res.json())
+      .then(listing => {
+        if (listing.length === 0) {
+          this.setState({ saved: false });
+        } else {
+          this.setState({ saved: true });
+        }
+      });
+  }
+
+  handleContactButton() {
+    fetch(`/api/listings/email/${this.props.listingId}`, {
+      method: 'GET'
+    })
+      .then(res => res.json())
+      .then(email => this.setState({ sellerEmail: email.email }))
+      .catch(err => console.error(err));
+    this.setState({
+      formActive: true
+    });
+  }
+
+  handleCancelButton() {
+    this.setState({
+      formActive: false
+    });
+  }
+
+  handleSaveButton() {
+    if (this.state.saved === false) {
+      fetch(`/api/user/saved/${this.props.listingId}`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-Access-Token': this.props.token
+        }
+      })
+        .then(res => {
+          this.setState({
+            saved: true,
+            savedPrompt: 'saved-prompt saved-view',
+            savedPromptText: 'Saved'
+          });
+        })
+        .catch(err => console.error(err));
+      this.prompt();
+    } else {
+      fetch(`/api/user/saved/${this.props.listingId}`, {
+        method: 'DELETE',
+        headers: {
+          'Content-Type': 'application/json',
+          'X-Access-Token': this.props.token
+        }
+      })
+        .then(res => {
+          this.setState({
+            saved: false,
+            savedPrompt: 'saved-prompt saved-view',
+            savedPromptText: 'Removed'
+          });
+        })
+        .catch(err => console.error(err));
+      this.prompt();
+    }
+  }
+
+  prompt() {
+    this.intervalID = setTimeout(() => {
+      this.setState({ savedPrompt: 'saved-prompt saved-hidden' });
+    }, 2000);
+  }
```

Comment:
As discussed via slack, you need a `componentWillUnmount()` lifecycle method that clears the timeout id so that you don't get the setState after unmount error.